.github/workflows/publish-and-deploy.yaml

```diff
@@ -0,0 +1,51 @@
+name: Publish and Deploy
+# Combined these two workflows for visbility
+
+on:
+  push:
+    branches: [ main ]
+
+jobs:
+  push_to_registry:
+    name: Push Docker Image to Docker Hub
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out the repo
+        uses: actions/checkout@v4
+      
+      - name: Log in to Docker Hub
+        uses: docker/login-action@f4ef78c080cd8ba55a85445d5b36e214a81df20a
+        with:
+          username: ${{ secrets.DOCKER_USERNAME }}
+          password: ${{ secrets.DOCKER_PASSWORD }}
+      
+      - name: Extract metadata (tags, labels) for Docker
+        id: meta
+        uses: docker/metadata-action@9ec57ed1fcdbf14dcef7dfbe97b2010124a938b7
+        with:
+          images: willnilges/meshdb
+      
+      - name: Build and push Docker image
+        uses: docker/build-push-action@3b5e8027fcad23fda98b2e3ac259d8d67585f671
```

Comment:
Does this publish the image publically? Are there any risks of credentials being "baked into" the image? I think we're using runtime environment variables for everything but just wanted to double check.

Support bot bakes in creds at build time I'm pretty sure (so we have to keep the builds private)